crates/uv-installer/src/compile.rs

```diff
@@ -88,6 +89,32 @@ pub async fn compile_tree(
     let tempdir = tempdir_in(cache).map_err(CompileError::TempFile)?;
     let pip_compileall_py = tempdir.path().join("pip_compileall.py");
 
+    let timeout: Option<Duration> = env::var(EnvVars::UV_COMPILE_BYTECODE_TIMEOUT)
+        .map(|value| {
+            // 0 indicates no timeout
+            if value == "0" {
+                None
+            } else {
+                Some(value.parse::<u64>()
+                    .map(Duration::from_secs)
+                    .or_else(|_| {
+                        // On parse error, warn and use the default timeout
+                        warn_user_once!("Ignoring invalid value from environment for `UV_COMPILE_BYTECODE_TIMEOUT`. Expected an integer number of seconds, got \"{value}\".");
```

Comment:
We should probably error here though, I don't think we have much precedent for warning on invalid options.